core/src/test/java/org/bitcoinj/core/DumpedPrivateKeyTest.java

```diff
@@ -66,7 +67,22 @@ public void cloning() throws Exception {
 
     @Test
     public void roundtripBase58() throws Exception {
-        String base58 = "5HtUCLMFWNueqN9unpgX2DzjMg6SDNZyKRb8s3LJgpFg5ubuMrk";
+        String base58 = "5HtUCLMFWNueqN9unpgX2DzjMg6SDNZyKRb8s3LJgpFg5ubuMrk";      // 32 bytes
         assertEquals(base58, DumpedPrivateKey.fromBase58(null, base58).toBase58());
     }
+
+    @Test
+    public void roundtripBase58_compressed() throws Exception {
+        String base58 = "cSthBXr8YQAexpKeh22LB9PdextVE1UJeahmyns5LzcmMDSy59L4";     // 33 bytes, compressed == true
+        assertEquals(base58, DumpedPrivateKey.fromBase58(null, base58).toBase58());
+    }
+
+    @Test(expected=AddressFormatException.class)
+    public void roundtripBase58_invalidCompressed() {
+        String base58 = "5Kg5shEQWrf1TojaHTdc2kLuz5Mfh4uvp3cYu8uJHaHgfTGUbTD";     // 33 bytes, compressed == false
+        byte[] bytes = Base58.decodeChecked(base58);
+        bytes = Arrays.concatenate(bytes, new byte[]{0});
```

Comment:
I wonder is there an alternative to using this spongycastle method?

Reply:
E.g. `com.google.common.primitives.Bytes.concat(byte[]...)`.

Reply:
Oh, sorry. I thought it was on java.util.Arrays. I'll change it.